protected/models/Patient.php

```diff
@@ -2159,12 +2159,17 @@ public function isCurrentlyInInterventionTrial($trial_id = null)
         return false;
     }
 
-    public function checkdate($attribute,$params)
+    public function dateOfBirthRangeValidator($attribute, $params)
     {
-        $currentdate = new DateTime(date("j M Y"));
+        if ($this->hasErrors('dob')) {
+            return;
+        }
```

Comment:
I'm not understanding why you are terminating the validator here. Shouldn't it run its course so ALL validation errors are displayed?

Reply:
if the user enters a completely invalid date of birth like "wasdasd", then an exception was being raised within this function because the date format was incorrect. I figured instead of duplicating the date format validation, we can just check to see if it has passed format validation before attempting to verify the range.